esmvaltool/utils/cmorizers/obs/cmorize_obs.py

```diff
@@ -0,0 +1,262 @@
+"""
+Run the CMORization module as a utility executable.
+
+This utility allows the user to call and execute CMOR reformatting
+scripts (support for NCL and Python at the moment), that will use
+two I/O variables passed by this utility: an input directory as
+specified in config-user.yml by the RAWOBS key, and an output dir
+created in the form of output_dir/CMOR_DATE_TIME/TierTIER/DATASET.
+The user can specify a list of DATASETS that the CMOR reformatting
+can by run on by using -o (--obs-list-cmorize) command line argument.
+The CMOR reformatting scripts are to be found in:
+esmvaltool/cmor/cmorizers/obs
+"""
+import argparse
+import logging
+import os
+import sys
+import datetime
+import subprocess
+
+from esmvaltool._task import write_ncl_settings
+from esmvaltool._config import read_config_user_file
+
+logger = logging.getLogger(__name__)
+
+HEADER = r"""
+______________________________________________________________________
+          _____ ____  __  ____     __    _ _____           _
+         | ____/ ___||  \/  \ \   / /_ _| |_   _|__   ___ | |
+         |  _| \___ \| |\/| |\ \ / / _` | | | |/ _ \ / _ \| |
+         | |___ ___) | |  | | \ V / (_| | | | | (_) | (_) | |
+         |_____|____/|_|  |_|  \_/ \__,_|_| |_|\___/ \___/|_|
+______________________________________________________________________
+
+""" + __doc__
+
+
+def _assemble_datasets(raw_obs, obs_list):
+    """Get my datasets as dictionary keyed on Tier."""
+    # check for desired datasets only (if any)
+    # if not, walk all over rawobs dir
+    # assume a RAWOBS/TierX/DATASET input structure
+    tiers = []
+    datasets = {}
+
+    # get all available tiers in source dir
+    for _, tier, _ in os.walk(raw_obs, followlinks=True):
```

Comment:
It would be better to be a bit more restrictive here, e.g.
```python
tiers = ['Tier{}'.format(i) for i in range(1, 4)]
tiers = [tier for tier in tiers if os.path.exists(os.path.join(raw_obs, tier))]
```

Reply:
But note that `Tier` ranges between 1 and 3.
Actually 2 and 3, since `Tier1` is `obs4mips` which is not supposed to be cmorized.